src/cloud/audiocom/ShareAudioDialog.cpp

```diff
@@ -69,105 +69,93 @@ const wxSize avatarSize = { 32, 32 };
 // A helper class that encapsulates exporting logic
 struct ExportHelper /* not final */
 {
-   AudacityProject& mProject;
-   wxString mPath;
-
-   explicit ExportHelper(AudacityProject& project)
-       : mProject { project }
-   {
-   }
+   virtual ~ExportHelper();
+
+   //! May modify global settings affecting export;
+   //! not responsible to revert changes
+   virtual void SetupUploadFormat() = 0;
```

Comment:
This interface was never intended for public use. It was solving very small and closed problem.